### DISCUSSION TITLE
Demands

### DISCUSSION THREAD

**Discussion by emersonramos** [kedacore/keda#4196] [General]
Hi,

I'm trying to use demands feature on Azure Pipelines on Keda 2.8, but I don't know where I'm going wrong.

I used demands on scaledobjects and scaledjobs and both produced this erros:

##[error]No agent found in pool keda which satisfies the specified demands: build, gradle, Agent.Version -gtVersion 2.172.0

I'm using this format on trigger:

triggers:
  - type: azure-pipelines
    metadata:
      poolName: "keda"
      organizationURLFromEnv: "AZP_URL"
      demands: "build,gradle"
    authenticationRef:
     name: pipeline-trigger-auth
     
I'm declaring the demands on pipeline too, using this format:

stages:
- stage: Build
  displayName: Build Application
  jobs:
  - job: Build
    displayName: Build Application
    pool:
      name: 'keda'
      demands:
        - "build"
        - "gradle"

It seems simple on doc, but it's not working on my environment.

I appreciate any help.

Tks

**Comment by JorTurFer** [comment]
Do you have at least one agent correctly registered in the pool? AzDO needs to have at least 1 agent (even if it's stopped) in the pool to schedule a job. Register an agent manually and that should be all
